src/main/java/com/iabtcf/model/TCModel.java

```diff
@@ -1,41 +1,85 @@
 package com.iabtcf.model;
 
-/*-
- * #%L
- * IAB TCF Core Library
- * %%
- * Copyright (C) 2020 IAB Technology Laboratory, Inc
- * %%
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- * 
- *      http://www.apache.org/licenses/LICENSE-2.0
- * 
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- * #L%
- */
-
 import java.time.Instant;
+import java.util.SortedSet;
+
+import com.iabtcf.model.ConsentLanguage;
+import com.iabtcf.model.PurposeRestriction;
+import com.iabtcf.model.PurposeRestrictionType;
 
 public interface TCModel {
 
+    ConsentLanguage static getConsentLanguages();
+
     /**
-     * @return the version of consent string format
+     * both of these are populated by decoding takes that value, otherwise creates timestamp of 'now'
      */
-    int version();
+    Instant getCreated();
+    Instant getLastUpdated();
 
     /**
-     * @return the {@link Instant} at which the consent string was created
+     * Each of these SortedSet<int> represents a bitfield in the TC String.  Keeping the set sorted is essential for
+     * encoding and using the SortedSet sublass of set creates a last() function which will return the highest id
+     * (this corresponds to the maxVendorID)
      */
-    Instant consentRecordCreated();
```

Comment:
With TCF v2, we've wholesale moved away from using the term "Consent" and moved to "Transparency and Consent" or just "TC".  Because the string now represents more than just the legal basis of "consent"